### PR TITLE
br: Check crypter.key valid before backup

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/29794

Problem Summary: 
the input `crypter.key` is invalid, the error information is come from TiKV. 
We can check the `crypter.key` valid on br directly. 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

---

% ./bin/br backup full --pd "172.16.6.117:2379" --storage "local:///tmp/backup/test" --crypter.method aes128-ctr --crypter.key 0123456789abcdef0123456789abcdefg
Error: encoding/hex: invalid byte: U+0067 'g'

% ./bin/br backup full --pd "172.16.6.117:2379" --storage "local:///tmp/backup/test" --crypter.method aes128-ctr --crypter.key 0123456789abcdef0123456789abcde
Error: encoding/hex: odd length hex string

% ./bin/br backup full --pd "172.16.6.117:2379" --storage "local:///tmp/backup/test" --crypter.method aes128-ctr --crypter.key 0123456789abcdef0123456789abcd
Error: crypter method and key not match: [BR:Common:ErrInvalidArgument]invalid argument

---

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/common_test.go [hunk 1]
```diff
@@ -62,3 +65,99 @@ func (s *testCommonSuite) TestStripingPDURL(c *C) {
...
+		},
+		{
+			CipherType: encryptionpb.EncryptionMethod_AES256_CTR,
+			CipherKey:  "",
+			ok:         false,
+		},
+	}
+
+	for _, t := range testCases {
+		cipherKey, err := hex.DecodeString(string(t.CipherKey))
```
Comment (kennytm) [inline br/pkg/task/common_test.go:116]:
```
[2021-11-22T07:06:22.511Z] /home/jenkins/agent/workspace/tidb_ghpr_check/go/src/github.com/pingcap/tidb/br/pkg/task/common_test.go:116:44: unnecessary conversion
```

File: br/pkg/task/common.go [hunk 2]
```diff
@@ -355,7 +355,7 @@ func (cfg *Config) parseCipherInfo(flags *pflag.FlagSet) error {
 	}
 
 	if !checkCipherKeyMatch(&cfg.CipherInfo) {
-		return errors.Annotate(err, "Cipher type and key not match")
+		return errors.Annotatef(berrors.ErrInvalidArgument, "crypter method and key not match")
```
Comment (kennytm) [inline br/pkg/task/common.go:358]:
```suggestion
		return errors.Annotate(berrors.ErrInvalidArgument, "crypter method and key length not match")
```

(or maybe keep using `Annotatef` but report the expected and actual length)

**Review by kennytm — APPROVED** [review]
rest lgtm

**Comment by fengou1** [comment]
LGTM

**Comment by kennytm** [comment]
@fengou1 please approve using this:

image

**Review by fengou1 — APPROVED** [review]

**Pull request merged**